File: C/Sorting_in_C/selection_sort.c
#include <stdio.h>
void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}
void selection(int array[], int length)
{
    int minIndex;
    for (int i = 0; i < length - 1; i++)
    {
        minIndex = i;
        for (int j = i; j < length; j++)
        {
            if (array[minIndex] > array[j])
            {
                swap(&array[minIndex], &array[j]);
            }
        }
    }
}
void print_Array(int array[], int length)
{
    for (int i = 0; i < length; i++)
    {
        printf("%d, ", array[i]);
    }
}
void main()
{
    int array[] = {1, 2, 4, 6, 0, 9, 8, 7, 6, 5};
    selection(array, 10);
    print_Array(array, 10);
}

File: Java/DSA/_Sorting_/src/SelectionSort.java
public class SelectionSort {
    void printArray(int[] array, int length) {
        for (int i = 0; i < length; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.println();
    }

    void Selection(int[] array, int length) {
        int minIndex = array[0], temp;
        for (int i = 0; i < length - 1; i++) {
            minIndex = i;
            for (int j = i + 1; j < length; j++) {
                if (array[j] < array[minIndex]) {
                    temp = array[j];
                    array[j] = array[minIndex];
                    array[minIndex] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] array = {1, 2, 3, 0, 9, 8, 7, 6, 5, 4};
        int length = 10;
        SelectionSort ss = new SelectionSort();
        ss.Selection(array, length);
        ss.printArray(array, length);
    }
}